src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -508,7 +456,7 @@ internal virtual bool DoRemove(T item)
                             // If parent is a 3-node, flip the orientation of the red link. 
                             // We can achieve this by a single rotation        
                             // This case is converted to one of other cased below.
-                            Debug.Assert(!parent.IsRed, "parent must be a black node!");
+                            Debug.Assert(!parent.IsRed);
```

Comment:
If it's not clear that `!IsRed` implies black in this case, I could introduce a property `IsBlack` and replace all `!IsRed` usages with that.